app/src/main/java/io/fabric/samples/cannonball/activity/PoemPopularActivity.java

```diff
@@ -61,37 +60,19 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_tweet_list);
 
         setUpViews();
-
-        loadTweets();
     }
 
     private void setUpViews() {
-        setUpBack();
         setUpPopularList();
+        setUpBack();
     }
 
     private void setUpPopularList() {
-        adapter = new TweetViewAdapter(PoemPopularActivity.this);
-        setListAdapter(adapter);
+        SearchTimeline searchTimeline = new SearchTimeline.Builder().query(SEARCH_QUERY).build();
 
+        final TweetTimelineListAdapter timelineAdapter = new TweetTimelineListAdapter(this, searchTimeline);
+        setListAdapter(timelineAdapter);
         getListView().setEmptyView(findViewById(R.id.loading));
```

Comment:
consider an empty view which has a retry button, which calls refresh on the adapter, in case the initial request fails.